client/js/widgets/scoreboard.js

```diff
@@ -0,0 +1,288 @@
+    else
+      rounds = [...Array(numRounds).keys()].map(i => i+1);
+    if(showTotals)
+      rounds.push('Totals');
+    rounds.unshift(this.get('roundLabel'));
+
+    // Finally, build the table
+    let tbl = document.createElement('table');
+    tbl.style.height = this.get('height') + 'px';
+    tbl.style.width = this.get('width')+ 'px';
+    let numRows;
+    let numCols;
+    let currentRound = parseInt(this.get('currentRound'));
+
+    if (isNaN(currentRound) || currentRound <1 || currentRound > numRounds)
+      currentRound = null;
+    
+    if(this.get('playersInColumns')) {
+      // Compute total number of rows and columns in table
+      numCols = pScores.length + 1;
+      numRows = numRounds + 1 + (showTotals ? 1 : 0);
+
+      // Add header row
+      const names = pScores.map(x => x[0]);
+      names.unshift(this.get('roundLabel'));
+      tbl.innerHTML += '<tbody></tbody>';
+      const tr = this.addRowToTable($('tbody', tbl), names, 'td');
+      if(usePlayerColors)
+        for (let c=0; c<pScores.length; c++ ) {
+          tr.cells[c+1].style.backgroundColor = colors[c];
+          tr.cells[c+1].style.color = 'white';
+        }
+      // Add remaining rows
+      for( let r=1; r < numRows; r++ ) {
+        const pRow = pScores.map(x => x[r]);
+        pRow.unshift(rounds[r]);
+        const tr = this.addRowToTable($('tbody',tbl), pRow, 'td');
+        tr.cells[0].classList.add('firstCol');
+        if(r == currentRound)
+          for(let c=1; c < numCols; c++)
+            tr.cells[c].classList.add('currentRound');
+      }
+      if(showTotals)
+          for(let c=0; c < numCols; c++)
+            tbl.rows[numRows-1].cells[c].classList.add('totalsLine');
+    } else { // Scores are in rows
+      // Compute total number of rows and columns in table
+      numCols = numRounds + 1 + (showTotals ? 1 : 0);
+      numRows = pScores.length + 1;
+
+      // First row contains round names
+      const tr = this.addRowToTable(tbl, rounds, 'td');
+      for (let c=0; c < numCols; c++)
+        tr.cells[c].classList.add('firstRow');
+      // Remaining rows are one row per player.
+      for( let r=0; r < pScores.length; r++) {
+        const tr = this.addRowToTable(tbl, pScores[r], 'td');
+        tr.cells[0].classList.add('firstCol');
+        if(usePlayerColors) {
+          tr.cells[0].style.backgroundColor = colors[r];
+          tr.cells[0].style.color = 'white';
+        }
+      }
+      for(let r=1; r < numRows; r++) {
+        if(currentRound)
+          tbl.rows[r].cells[currentRound].classList.add('currentRound');
+        if(showTotals)
+          tbl.rows[r].cells[numCols-1].classList.add('totalsLine');
+      }
+    }
+
+    // Make top row, left column non-scrollable
+    for (let r=0; r<numRows; r++) 
+      tbl.rows[r].cells[0].classList.add('firstCol');
+    for (let c=0; c< numCols; c++)
+      tbl.rows[0].cells[c].classList.add('firstRow');
+
+    const myDom = this.domElement;
+    myDom.innerHTML = '';
```

Comment:
This is deleting DOM elements of child widgets.

Reply:
Ack. So what's the right way to fix this? I guess the table needs to have an ID that is determinable from the scoreboard ID; then it can be removed and re-added.